redis/rediscache.go

```diff
@@ -0,0 +1,39 @@
+package rediscache
+
+import "github.com/garyburd/redigo/redis"
+
+// Cache is an implementation of httpcache.Cache that caches responses in a
+// redis server.
+type Cache struct {
+	redis.Conn
+}
+
+// cacheKey modifies an httpcache key for use in redis.  Specifically, it
+// prefixes keys to avoid collision with other data stored in redis.
+func cacheKey(key string) string {
+	return "rediscache:" + key
+}
+
+// Get returns the response corresponding to key if present.
+func (c Cache) Get(key string) (resp []byte, ok bool) {
+	item, err := redis.Bytes(c.Do("GET", cacheKey(key)))
+	if err != nil {
+		return nil, false
+	}
+	return item, true
+}
+
+// Set saves a response to the cache as key.
+func (c Cache) Set(key string, resp []byte) {
+	c.Do("SET", cacheKey(key), resp)
+}
+
+// Delete removes the response with key from the cache.
+func (c Cache) Delete(key string) {
+	c.Do("DEL", cacheKey(key))
+}
+
+// NewWithClient returns a new Cache with the given redis connection.
+func NewWithClient(client redis.Conn) Cache {
+	return Cache{client}
+}
```

Comment:
I know you did it like this because it's how the other cache packages are, but I think we take this opportunity to potentially improve.

Right now, there are two ways to create a new cache:

```Go
c := rediscache.NewWithClient(redisConn)
```

Or:

```Go
c := rediscache.Cache{Conn: redisConn}
```

I think it's unhelpful. It makes the API larger, and makes it confusing having to choose one of two ways to do something when there's no difference. It creates divergence is style as some people will pick option 1 while others option 2.

What if we only provided one way, and took the other way out. Wouldn't that be better?

In this case, a redis client is mandatory, the `rediscache.Cache` is useless without one. So we could make `cache` unexported and make `NewWithClient` the only way to create a client.

What do you think?

(If you'd rather not do this, we can leave it out of this PR and deal with this issue separately later.)